File: LocalizationEditor/Localization/ClientApp/src/app/connection/connection-data/connection.ts
import {IConnection} from "./IConnection";
import {EntityState} from "../../base/entity-state";

export class Connection implements IConnection{
  id: number;
  name: string;
  connectionString: string;
  state: EntityState;

  constructor(id: number, name: string, connection: string) {
    this.id = id;
    this.name = name;
    this.connectionString = connection;
  }

  isNewEntity():boolean {
    return this.id == 0;
  }

  setState(state: EntityState): void{
    this.state = state;
  }
}

File: LocalizationEditor/ConnectionString/Models/Implementation/Connection.cs
namespace ConnectionString.Models.Implementation
{
  internal class Connection : IConnection
  {
    public long Id { get; }
    public string Name { get; private set; }
    public string ConnectionString { get; private set; }

    public Connection(
      long id,
      string name,
      string connectionString)
    {
      Id = id;
      Name = name;
      ConnectionString = connectionString;
    }

    public void Update(IConnection connection)
    {
      Name = connection.Name;
      ConnectionString = connection.ConnectionString;
    }
  }
}